CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKImage.java

```diff
@@ -455,28 +600,353 @@ public static CVKImage Create(  CVKDevice cvkDevice,
-            vkViewInfo.format(cvkImage.GetFormat());
-            vkViewInfo.subresourceRange().aspectMask(cvkImage.GetAspectMask());
-            vkViewInfo.subresourceRange().baseMipLevel(0);
-            vkViewInfo.subresourceRange().levelCount(1);
-            vkViewInfo.subresourceRange().baseArrayLayer(0);
-            vkViewInfo.subresourceRange().layerCount(layers);
-
-            ret = vkCreateImageView(cvkDevice.GetDevice(), vkViewInfo, null, cvkImage.pImageView);
-            checkVKret(ret);           
-            
-            return cvkImage;
+        }
+        return cvkImage;
+    }
+
+
+    /**
+     * Creates an ImageView associate with the image handle
+     * @return error code
+     */
+    public int CreateImageView() {
+        int ret = VK_SUCCESS;
+        
+        try(MemoryStack stack = stackPush()) {
+                // Create an image view that can be used to read and write this image
+                VkImageViewCreateInfo vkViewInfo = VkImageViewCreateInfo.callocStack(stack);
+                vkViewInfo.sType(VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO);
+                vkViewInfo.image(GetImageHandle());
+                vkViewInfo.viewType(viewType);
+                vkViewInfo.format(GetFormat());
+                vkViewInfo.subresourceRange().aspectMask(GetAspectMask());
+                vkViewInfo.subresourceRange().baseMipLevel(0);
+                vkViewInfo.subresourceRange().levelCount(1);
+                vkViewInfo.subresourceRange().baseArrayLayer(0);
+                vkViewInfo.subresourceRange().layerCount(layers);
+
+                ret = vkCreateImageView(cvkDevice.GetDevice(), vkViewInfo, null, pImageView);
+                checkVKret(ret);
+
         } catch (Exception e) {
-            //TODO_TT: move this to class destructor
-            cvkImage.Destroy();
+            // TODO HYDRA TEST THIS CODE PATH       
+            cvkDevice.GetLogger().info("Image View creation failed for Image Handle %d. Image will be destroyed!", GetImageHandle());
+            ret = CVK_ERROR_IMAGE_VIEW_CREATION_FAILED;
+            
+            DestroyImage();
+            DestroyImageMemory();
         }
         
-        return null;
+        return ret;
+    }
+    
+    
+    /**
+     * Prepares the image 
+     * 
+     * @param stack
+     * @return 
+     */
+    public CVKImage PrepareImageToMemory(MemoryStack stack) {
+        int ret = VK_SUCCESS;
+ 	boolean supportsBlit = true;
+        int originalLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+                
+        // Check blit support for source and destination
+        // TODO HYDRA do this once and store result in cvkdevice?
+        VkFormatProperties formatProps = VkFormatProperties.callocStack(stack);
+
+        // Check if the device supports blitting from optimal images (the swapchain images are in optimal format)
+        vkGetPhysicalDeviceFormatProperties(cvkDevice.GetPhysicalDevice(), format, formatProps);
```

Comment:
Can you move this to CVKDevice please?  We already have checks very similar to this there.

Reply:
Yes I already have a TODO in there to do that